### PR TITLE
fixed incorrect exception mapping

### PR DESCRIPTION
**Pull request by davilu**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: e2e/test/MessageReceiveE2ETests.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ public static async Task VerifyReceivedC2DMessageAsync(Client.TransportType tran
                 }
                 else
                 {
-                    receivedMessage = await dc.ReceiveAsync(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+                    receivedMessage = await dc.ReceiveAsync(TimeSpan.FromSeconds(10)).ConfigureAwait(false);
                 }
```
Comment (prmathur-microsoft) [inline e2e/test/MessageReceiveE2ETests.cs]: Why do we need to wait 10 sec now?

File: iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public class MqttTransportSettings : ITransportSettings
         const int DefaultMaxPendingInboundMessages = 50;
         const QualityOfService DefaultPublishToServerQoS = QualityOfService.AtLeastOnce;
         const QualityOfService DefaultReceivingQoS = QualityOfService.AtLeastOnce;
-        static readonly TimeSpan DefaultConnectArrivalTimeout = TimeSpan.FromSeconds(300);
+        static readonly TimeSpan DefaultConnectArrivalTimeout = TimeSpan.FromSeconds(10);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs]: Clarify from service about the avg time taken to ack connect packet

File: iothub/device/src/Transport/RetryDelegatingHandler.cs [hunk 3]
```diff
@@ -189,9 +189,9 @@ public override async Task EnableMethodsAsync(CancellationToken cancellationToke
                 {
                     await EnsureOpenedAsync(cancellationToken).ConfigureAwait(false);
 
+                    await _handlerLock.WaitAsync(cancellationToken).ConfigureAwait(false);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/RetryDelegatingHandler.cs]: Why is `_handlerLock` moved out of the `try` block?

**Pull request merged**